### PR TITLE
New Components - dust

### LINKED ISSUE #14021: [Components] dust
# dust
## URLs
- https://docs.dust.tt/reference/developer-platform-overview
#### Prompt
Send a message to an assistant on Dust and receive an answer. The message content is the required prop.
#### URLs
- https://docs.dust.tt/reference/get_api-v1-w-wid-assistant-agent-configurations
- https://docs.dust.tt/reference/post_api-v1-w-wid-assistant-conversations
#### Prompt
Add a document to a chosen Dust Data Source or Folder. Two required props for this action are the document to be uploaded and the destination Data Source or Folder.
#### URLs
- https://docs.dust.tt/reference/get_api-v1-w-wid-data-sources
- https://docs.dust.tt/reference/post_api-v1-w-wid-data-sources-name-documents-documentid

### PR DESCRIPTION
**Pull request by luancazarine**
Resolves #14021.

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Talk Assistant" action for user interaction with an assistant on the Dust platform.
	- Added an "Upsert Document" action to update or insert documents into a specified data source.

- **Improvements**
	- Enhanced the Dust application module with new properties and methods for better API interactions.

- **Version Update**
	- Updated the version of the Dust component to 0.1.0 and added new dependencies.

### CODE REVIEW

File: components/dust/actions/talk-assistant/talk-assistant.mjs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        dust,
+        "assistantId",
+      ],
+    },
+    content: {
+      propDefinition: [
+        dust,
+        "content",
+      ],
+    },
```
Comment (michelle0927) [inline components/dust/actions/talk-assistant/talk-assistant.mjs:23]: The prop `content` is not present in the app file.

File: components/dust/actions/upsert-document/upsert-document.mjs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      $,
+      dataSourceId: this.dataSourceId,
+      documentId: this.documentId,
+      data: {
+        text: this.content,
+        light_document_output: this.lightDocumentOutput,
+      },
+    });
+
+    $.export("$summary", `Successfully uploaded document to destination ${this.destination}`);
```
Comment (michelle0927) [inline components/dust/actions/upsert-document/upsert-document.mjs:46]: `this.destination` doesn't exist in the component.

File: components/dust/dust.app.mjs [hunk 3]
```diff
@@ -1,11 +1,90 @@
...
+          label: `${name} - ${description}`,
+          value,
+        }));
+      },
+    },
+    document: {
+      type: "string",
+      label: "Document",
+      description: "The content of the document to be uploaded",
+    },
```
Comment (michelle0927) [inline components/dust/dust.app.mjs:41]: The prop `document` isn't being used anywhere.
```suggestion
```

**Review by michelle0927 — CHANGES_REQUESTED** [review]

**Review by michelle0927 — APPROVED** [review]
LGTM!

**Review by michelle0927 — APPROVED** [review]

**Comment by spolu** [comment]
:exploding_head:

**Pull request merged**